File: CSES/increasingArray.java
import java.io.*;
import java.util.StringTokenizer;
//ignore old java code
public class increasingArray {
    public static void main(String args[]) throws IOException
    {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(System.out);
        StringTokenizer st = new StringTokenizer(r.readLine());

        int test = Integer.parseInt(st.nextToken());
        long moves = 0;
        //code added so vscode stops yelling at me
        moves += test;
        moves -= test;
        //done
        st = new StringTokenizer(r.readLine());
        int last = Integer.parseInt(st.nextToken());
        int curr = 0;

        while (st.hasMoreTokens())
        {
            curr = Integer.parseInt(st.nextToken());
            if (curr < last)
            {
                moves += last - curr;
                curr = last;
            }
            last = curr;
        }
        pw.println(moves);

        pw.close();
    }
}

File: CSES/increasingArray.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    cin.tie(0)->sync_with_stdio(0);

    int n, last, curr = 0;
    cin >> n >> last;

    long moves = 0;

    for (int i = 1; i < n; i++)
    {
        cin >> curr;
        if (curr < last)
        {
            moves += last - curr;
            curr = last;
        }
        last = curr;
    }
    cout << moves << "\n";
}